### ISSUE TITLE
Weird memory leak behavior

### ISSUE THREAD

**Issue by cyoon1729** [cyoon1729/RLcycle#11]
DQN-based algorithms have memory leak while training on Atari. The leak doesn't always happen (inconsistent), or happens suddenly in the middle of training.

**Comment by cyoon1729** [comment]
The function below (numpy to tensor conversion) causes the memory leak:

```python
"""in rlcycle/common/utils/common_utils.py"""
def np2tensor(np_arr: np.ndarray, device: torch.device):
    """Convert numpy array to tensor"""
    tensor_output = torch.FloatTensor(np_arr).to(device)
    if device.type == "cuda":
        tensor_output.cuda(non_blocking=True)
    return tensor_output


def np2tensor2(np_arr: np.ndarray, device: torch.device):
    """Convert numpy array to tensor"""
    tensor_output = torch.from_numpy(np_arr).float().to(device)
    if device.type == "cuda":
        tensor_output.cuda(non_blocking=True)
    return tensor_output
```

**Comment by cyoon1729** [comment]
Fixed memory leak in PR #12 
Here's what changed:
1. Numpy to tensor conversion:
```python
def np2tensor(np_arr: np.ndarray, use_cuda: bool):
    """Convert numpy array to tensor"""
    if use_cuda:
        return torch.from_numpy(np_arr).cuda(non_blocking=True).float()
    return torch.from_numpy(np_arr).cpu().float()
```
This was recommended by Pytorch in [this](https://discuss.pytorch.org/t/converting-numpy-array-to-tensor-on-gpu/19423) discussion thread.

2. Use `cuda()` instead `.to(device)' to cast cuda tensors. Recommended by Pytorch in [this](https://discuss.pytorch.org/t/converting-numpy-array-to-tensor-on-gpu/19423/4)  discussion thread.

3. Added a [memory profiler](https://github.com/cyoon1729/RLcycle/blob/master/rlcycle/common/utils/debug/memory.py) for debugging.
